yt_dlp/extractor/digiteka.py

```diff
@@ -25,74 +25,83 @@ class DigitekaIE(InfoExtractor):
-            'duration': 233,
-            'upload_date': '20150224',
-            'timestamp': 1424760500,
-            'uploader_id': '3rfzk',
-        },
-    }, {
-        'url': 'https://www.digiteka.net/deliver/generic/iframe/mdtk/01637594/src/lqm3kl/zone/1/showtitle/1/autoplay/yes',
-        'only_matching': True,
+        'params': {'skip_download': True},
     }]
 
     def _real_extract(self, url):
         mobj = self._match_valid_url(url)
         video_id = mobj.group('id')
-        video_type = mobj.group('embed_type') or mobj.group('site_type')
-        if video_type == 'music':
-            video_type = 'musique'
 
-        deliver_info = self._download_json(
-            f'http://www.ultimedia.com/deliver/video?video={video_id}&topic={video_type}',
-            video_id)
+        IFRAME_MD_ID = '01836272'   # One static ID working for Ultimedia iframes
 
-        yt_id = deliver_info.get('yt_id')
-        if yt_id:
-            return self.url_result(yt_id, 'Youtube')
+        api_url = f'https://www.ultimedia.com/player/getConf/{IFRAME_MD_ID}/1/{video_id}'
+
+        conf_data = self._download_json(
+            api_url, video_id, note='Downloading player configuration')
 
-        jwconf = deliver_info['jwconf']
+        video_info = conf_data.get('video')
+        if not video_info:
+            raise ExtractorError('Failed to retrieve video information from API.', expected=True)
 
+        title = video_info['title']
         formats = []
-        for source in jwconf['playlist'][0]['sources']:
+        media_sources = video_info.get('media_sources', {})
+
+        hls_sources = media_sources.get('hls', {})
+        for format_id, hls_url in hls_sources.items():
+            if not hls_url:
+                continue
+
+            height_str = format_id.split('_')[-1]
+            height = int(height_str) if height_str.isdigit() else None
+
             formats.append({
-                'url': source['file'],
-                'format_id': source.get('label'),
+                'url': hls_url,
+                'format_id': f'hls-{height_str}',
+                'height': height,
+                'protocol': 'm3u8_native',
+                'ext': 'mp4',
             })
 
-        title = deliver_info['title']
-        thumbnail = jwconf.get('image')
-        duration = int_or_none(deliver_info.get('duration'))
-        timestamp = int_or_none(deliver_info.get('release_time'))
-        uploader_id = deliver_info.get('owner_id')
+        mp4_sources = media_sources.get('mp4', {})
+        for format_id, mp4_url in mp4_sources.items():
+            if not mp4_url:
+                continue
+
+            height_str = format_id.split('_')[-1]
+            height = int(height_str) if height_str.isdigit() else None
+
+            formats.append({
+                'url': mp4_url,
+                'format_id': f'mp4-{height_str}',
+                'height': height,
+                'ext': 'mp4',
+            })
```

Comment:
I would write this like so
```suggestion
        for format_id, mp4_url in traverse_obj(video_info, ('media_sources', 'mp4', {dict.items})):
            if not mp4_url:
                continue

            formats.append({
                'url': mp4_url,
                'format_id': format_id,
                'height': int_or_none(format_id.partition('_')[2]),
                'ext': 'mp4',
            })
```